### PR TITLE
options button test

### PR DESCRIPTION
**Pull request by Alexsv7**

### CODE REVIEW

**Comment by vpodk** [comment]
Please add success criteria description.

File: tests/testoptionsbutton.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  toggle.dispatchEvent(new Event('click'));
+
+  // Check if after second click on toggle all expanded widgets was collapsed.
+  if (expanded.length) {
+    return true;
+  }
+
+  return false;
```
Comment (vpodk) [inline tests/testoptionsbutton.js]:
``` javascript
/** @type {boolean} */ var result = false;
...
return result; // Only one return statement.
```

File: tests/testoptionsbutton.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if (buttonClosed) {
+      if (optionsButton.dispatchEvent(new Event('click'))) {
+        if (toggleButton) {
+          toggleButton.dispatchEvent(new Event('click'));
+          /** @type {Array|NodeList} */
+          var expanded = document.getElementsByClassName('expanded');
+          if (expandable.length = expanded.length) {
+            toggleButton.dispatchEvent(new Event('click'));
+            if (!expanded.length) {
+              return false;
```
Comment (vpodk) [inline tests/testoptionsbutton.js]:
``` javascript
/** @type {boolean} */ var result = false;
...
return result; // Only one return statement.
```

Only one return statement. This means that one (`1`) is greater than zero (`0`) and less than two (`2`).

File: tests/testoptionsbutton.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  var toggleButton = document.getElementById('toggle-widgets-view');
+  /** @type {Array|NodeList} */
+  var expandable = document.getElementsByClassName('is-expandable');
+  /** @type {Array|NodeList} */ var expanded;
+
+  // Check if options button exists.
+  if (optionsButton) {
+    if (buttonClosed) {
+      if (optionsButton.dispatchEvent(new Event('click'))) {
+        if (toggleButton) {
```
Comment (vpodk) [inline tests/testoptionsbutton.js]:
``` javascript
if (optionsButton && buttonClosed && toggleButton) {
...

```

File: tests/testoptionsbutton.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          toggleButton.dispatchEvent(new Event('click'));
+          /** @type {Array|NodeList} */
+          var expanded = document.getElementsByClassName('expanded');
+          if (expandable.length = expanded.length) {
```
Comment (vpodk) [inline tests/testoptionsbutton.js]: WTF?

File: tests/testoptionsbutton.js (hunk 3 above)
Comment (RomanFedytskyi) [inline tests/testoptionsbutton.js]: fixed

File: tests/testoptionsbutton.js (hunk 2 above)
Comment (RomanFedytskyi) [inline tests/testoptionsbutton.js]: fixed

File: tests/testoptionsbutton.js (hunk 4 above)
Comment (RomanFedytskyi) [inline tests/testoptionsbutton.js]: fixed

File: tests/testoptionsbutton.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  /** @type {Element} */
+  var toggleButton = document.getElementById('toggle-widgets-view');
+  /** @type {Array|NodeList} */
+  var expandable = document.getElementsByClassName('is-expandable');
+  /** @type {Array|NodeList} */ var expanded;
+
+  if (optionsButton && buttonClosed && toggleButton) {
+    if (optionsButton.dispatchEvent(new Event('click'))) {
+      toggleButton.dispatchEvent(new Event('click'));
+      /** @type {Array|NodeList} */
```
Comment (mshchygol) [inline tests/testoptionsbutton.js]: unnecessary var declaration

File: tests/testoptionsbutton.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  var expandable = document.getElementsByClassName('is-expandable');
+  /** @type {Array|NodeList} */ var expanded;
+
+  if (optionsButton && buttonClosed && toggleButton) {
+    if (optionsButton.dispatchEvent(new Event('click'))) {
+      toggleButton.dispatchEvent(new Event('click'));
+      expanded = document.getElementsByClassName('expanded');
+      if (expanded) {
+        toggleButton.dispatchEvent(new Event('click'));
+        if (!expanded.lengt) {
```
Comment (RomanFedytskyi) [inline tests/testoptionsbutton.js]: length

**Comment by mshchygol** [comment]
LGTM 👍

**Pull request closed without merging**